outputs/jira_test.go

```diff
@@ -0,0 +1,1074 @@
+		content        map[string]string
+		wantError      string
+	}{
+		{
+			name: "happy path",
+			jiraApi: &JiraAPI{
+				ProjectKey:      "project",
+				User:            "user",
+				boardType:       "scrum",
+				Labels:          []string{"label1", "label2"},
+				FixVersions:     []string{"fix1", "fix2"},
+				AffectsVersions: []string{"affect1", "affect2"},
+			},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			fieldList: &[]jira.Field{
+				{ID: "issuetype", Name: "Issue Type"},
+				{ID: "project", Name: "Project"},
+				{ID: "priority", Name: "Priority"},
+				{ID: "description", Name: "Description"},
+				{ID: "summary", Name: "Summary"},
+			},
+			priorityList: &[]jira.Priority{{Name: "High"}},
+			issue:        &jira.Issue{},
+			content:      map[string]string{"title": "title_content", "description": "description_content"},
+		},
+		{
+			name:      "sad path (Failed to create client)",
+			wantError: "Failed to create client",
+		},
+		{
+			name:      "sad path (Failed to create meta project)",
+			jiraApi:   &JiraAPI{},
+			wantError: "Failed to create meta project",
+		},
+		{
+			name:           "sad path (Failed to create issuetype)",
+			jiraApi:        &JiraAPI{Issuetype: "Bug", ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			wantError:      "Failed to create issuetype",
+		},
+		{
+			name:           "sad path (Failed to create fields config)",
+			jiraApi:        &JiraAPI{ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			wantError:      "Failed to create fields config",
+		},
+		{
+			name:           "sad path (Failed to init issue)",
+			jiraApi:        &JiraAPI{ProjectKey: "project", Unknowns: map[string]string{"bad field": "bad field"}},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			wantError:      "key bad field is not found in the list of fields",
+		},
+		{
+			name:           "sad path (Failed to open issue)",
+			jiraApi:        &JiraAPI{ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			wantError:      "Failed to open issue",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mux := http.NewServeMux()
+			mux.HandleFunc("/rest/api/2/issue/createmeta", func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError == "Failed to create meta project" {
```

Comment:
as code is almost same I would create something like `buildHttpHandler(successResponse, errorResponse) func ....`

Reply:
Done